mlflow/store/tracking/sqlalchemy_store.py

```diff
@@ -1548,6 +1549,11 @@ def start_trace(
             )
 
             trace_info.tags = [SqlTraceTag(key=k, value=v) for k, v in tags.items()]
+            trace_info.tags.append(
```

Comment:
note: This is not related to the main change, but we have to set this tag in SQL store as well, to allow downloading trace data.

Reply:
This seems not the correct place to upload/download trace data. We put trace_data in {experiment_artifact_location}/traces/{request_id}/artifacts in file_store, we might want to do the same for sql store? Since we have other run artifacts in {experiment_artifact_location} we might want to isolate the two.

Reply:
I see, I mimiced Databricks behavior (where it simply uses the run artifact URI), but isolating them makes sense as we don't use the same table for run/trace. Will update!

Reply:
Actually let me split out this artifact location piece to a separate PR. It will make it easy to revert the DB migration change in case of any issue😅